standards/SDS-4-Orbit-Parameter-Message-(OPM)/main.fbs

```diff
@@ -0,0 +1,154 @@
+  MET,
+  /// Mission Relative Time
+  MRT,
+  /// Spacecraft Clock (receiver) (requires rules for interpretation in ICD)
+  SCLK,
+  /// International Atomic Time
+  TAI,
+  /// Barycentric Coordinate Time
+  TCB,
+  /// Barycentric Dynamical Time
+  TDB,
+  /// Geocentric Coordinate Time
+  TCG,
+  /// Terrestrial Time
+  TT,
+  /// Universal Time
+  UT1,
+  /// Coordinated Universal Time 
+  UTC
+}
+
+enum manCovRefFrame : byte {
+  // Another name for ‘Radial, Transverse, Normal’
+  RSW,
+  // Radial, Transverse, Normal
+  RTN,
+  // A local orbital coordinate frame (x velocity vec., w orbital angular momentum vec., N completes right handed system.)
+  TNW
+}
+
+table maneuverParameters {
+  // Maneuver Parameters (Repeat for each maneuver. None or all parameters of this block must be given.)
+  MAN_EPOCH_IGNITION:string;
+  MAN_DURATION:double;
+  MAN_DELTA_MASS:double;
+  MAN_REF_FRAME:manCovRefFrame;
+  MAN_DV_1:double;
+  MAN_DV_2:double;
+  MAN_DV_3:double;
+}
+
+table OPM {
+  // OPM Header
+  CCSDS_OPM_VERS:double = null;
+  CREATION_DATE:string;
+  ORIGINATOR:string;
+
+  // OPM Metadata
+  OBJECT_NAME:string;
+  OBJECT_ID:string;
+  CENTER_NAME:string;
+  REF_FRAME:referenceFrame;
+  REF_FRAME_EPOCH:string;
+  TIME_SYSTEM:timeSystem;
+  
+  // State Vector Components in the Specified Coordinate System
+  COMMENT:string;
+  EPOCH:string;
+  X:double;
+  Y:double;
+  Z:double;
+  X_DOT:double;
+  Y_DOT:double;
+  Z_DOT:double;
+
+  // Osculating Keplerian Elements in the Specified Reference Frame (none or all parameters of this block must be given.)
+  SEMI_MAJOR_AXIS:double = null;
+  ECCENTRICITY:double = null;
+  INCLINATION:double = null;
+  RA_OF_ASC_NODE:double = null;
+  ARG_OF_PERICENTER:double = null;
+  TRUE_ANOMALY:double = null;
+  MEAN_ANOMALY:double = null;
+  GM:double = null;
+  
+  // Spacecraft Parameters
+  MASS:double = null;
+  SOLAR_RAD_AREA:double = null;
+  SOLAR_RAD_COEFF:double = null;
+  DRAG_AREA:double = null;
+  DRAG_COEFF:double = null;
+
+  // Position/Velocity Covariance Matrix (6x6 Lower Triangular Form. None or all parameters of the matrix must be given.
+  // COV_REF_FRAME may be omitted if it is the same as the metadata REF_FRAME.) 
+  COV_REF_FRAME:manCovRefFrame;
+  CX_X:double = null;
+  CY_X:double = null;
+  CY_Y:double = null;
+  CZ_X:double = null;
+  CZ_Y:double = null;
+  CZ_Z:double = null;
+  CX_DOT_X:double = null;
+  CX_DOT_Y:double = null;
+  CX_DOT_Z:double = null;
+  CX_DOT_X_DOT:double = null;
+  CY_DOT_X:double = null;
+  CY_DOT_Y:double = null;
+  CY_DOT_Z:double = null;
+  CY_DOT_X_DOT:double = null;
+  CY_DOT_Y_DOT:double = null;
+  CZ_DOT_X:double = null;
+  CZ_DOT_Y:double = null;
+  CZ_DOT_Z:double = null;
+  CZ_DOT_X_DOT:double = null;
+  CZ_DOT_Y_DOT:double = null;
+  CZ_DOT_Z_DOT:double = null;
+
+  MANEUVERS:[maneuverParameters];
+
+  // User Defined Parameters (all parameters in this section must be described in an ICD). 
+  USER_DEFINED_BIP_0044_TYPE:uint;
```

Comment:
@TJKoury - I copied these from the OMM schema, but I'm not sure if they all apply. The CCSDS doc only gives earth_model as an example, so I'm not sure what other use cases you had in mind for others. Let me know your thoughts.

Reply:
USER_DEFINED_OBJECT_DESIGNATOR: this is for internal IDs used by organizations / users that are non-standard / conforming.

USER_DEFINED_EARTH_MODEL: per the CCSDS documentation.

USER_DEFINED_EPOCH_TIMESTAMP: Useful when storing date values as a UNIX timestamp, to remove parsing requirement, which is something FlatBuffers is supposed to enable

USER_DEFINED_MICROSECONDS: Several data providers (such as Space-Track) provide microsecond precision for timestamps.  Some language parsers have issues with ISO8601 dates with microsecond precision, so the option is to store the microsecond portion of the EPOCH in this user-defined field.

Might be a good idea to rename it to `USER_DEFINED_EPOCH_MICROSECONDS` or similar for specificity.

Reply:
Name updated.